kornia/contrib/models/rt_detr/model.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from enum import Enum
+
+from kornia.contrib.models import DetectionResults
+from kornia.contrib.models.base import ModelBase
+from kornia.contrib.models.rt_detr.architecture.hybrid_encoder import HybridEncoder
+from kornia.contrib.models.rt_detr.architecture.resnet_d import ResNetD
+from kornia.contrib.models.rt_detr.architecture.rtdetr_head import RTDETRHead
+from kornia.core import Tensor, concatenate, tensor
+
+
+class RTDETRModelType(Enum):
+    r50 = 0
+    r101 = 1
+    l = 2  # noqa: E741
+    x = 3
+
+
+@dataclass
+class RTDETRConfig:
+    model_type: str | int | RTDETRModelType | None = None
+    checkpoint: str | None = None
+
+
+class RTDETR(ModelBase[RTDETRConfig]):
+    def __init__(self, backbone: ResNetD, neck: HybridEncoder, head: RTDETRHead):
+        super().__init__()
+        self.backbone = backbone
+        self.neck = neck
+        self.head = head
+
+    @staticmethod
+    def from_config(config: RTDETRConfig) -> RTDETR:
+        model_type = config.model_type
+
+        if isinstance(model_type, int):
+            model_type = RTDETRModelType(model_type)
+        elif isinstance(model_type, str):
+            model_type = getattr(RTDETRModelType, model_type)
```

Comment:
in this case, maybe is fair to add a `.get` method on the Enum class, to cover it. Similar to https://github.com/kornia/kornia/blob/master/kornia/constants.py#L87 or https://github.com/kornia/kornia/blob/master/kornia/constants.py#L66

Reply:
Does it make sense to also add a .compile function to wrap torch.compile?

Reply:
@edgarriba `ModelBase` class already defines that?

Reply:
you are right, just remembered that the visual prompter (Sam) we had to explicitly compile some parts of the model. Worth to add a test with .compile() to verify that at least doesn't crash

Reply:
I do include a dynamo test. The .compile() does not crash. On my Mac M1 CPU, without low_tolerance=True, sometimes the test fails with very low % mismatched (e.g. 1/40 or 2.5%). With low tolerance, the test always passes.

Reply:
Great, but I’d add the test in any case at least to see the pass with dummy data.
Have you also tried to export to ONNX, that’s I think also next step for models so that they are exportable.

Reply:
Yes, the dynamo test is included in `test_rt_detr.py`

About ONNX export, I just tried it out. There are 2 problems:

1. ONNX export will fail with our custom data structure DetectionResults. The solution is to have a wrapper (or an internal method) that returns a list of tensors (class id, scores, bboxes) or a packed tensor instead (class id, scores, bboxes for each sample)
2. aten::scaled_dot_product_attention cannot be exported directly to ONNX. Related issue from PyTorch core: https://github.com/pytorch/pytorch/issues/97262

Reply:
For ONNX, I’m reading that there are some potential tricks to apply like scaling before . Have you tried those?